### PR TITLE
Get event crew

### LINKED ISSUE #92: getNotEventCrewUsers
- Endpoint URI: */users/event-crews/{eventId}*
- Http Method: *GET*
- API Logic: 
    - find all existing users
    - filter out users that is not the existing event crew of event with "eventId"
    - return to client
    
- Progress
    - [ ] development
    - [ ] integration Test

### LINKED ISSUE #94: getEventCrewWithEventId
- Endpoint URI: */event-crews/event/{eventId}*
- Http Method: *GET*
- API Logic: 
    - get all event crew from eventId
    - add user details to the the event crew dto
    
- Progress
    - [ ] development
    - [ ] integration Test

### PR DESCRIPTION
**Pull request by junzai97**
- Resolved #92
- Resolve #94 
- Update EventCrewDTO
- Update EventCrew findOne logic
- Add pre-authorize to EventCrew Endpoint

### CODE REVIEW

File: src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java [hunk 1]
```diff
@@ -74,7 +108,10 @@ public EventCrewDTO save(EventCrewDTO eventCrewDTO) {
     public Optional<EventCrewDTO> findOne(Long id) {
         log.debug("Request to get EventCrew : {}", id);
         return eventCrewRepository.findById(id)
-            .map(eventCrewMapper::toDto);
+            .map(eventCrewMapper::toDto)
+            .map(this::mapUserDetails)
+            .map(this::mapEventName)
+            .map(this::mapUserDetails);
```
Comment (CarlosSia) [inline src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java]: why here mapUserDetails for second time de?

File: src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java [hunk 2]
```diff
@@ -85,6 +122,38 @@ public EventCrewDTO save(EventCrewDTO eventCrewDTO) {
...
+    /**
+     * Assign user details(name and contact number) to the eventCrewDTO.
+     *
+     * @param eventCrewDTO to add on the user details' prop.
+     */
+    private EventCrewDTO mapUserDetails(EventCrewDTO eventCrewDTO) {
+        Optional<User> dbUser = userRepository.findById(eventCrewDTO.getUserId());
+        if(dbUser.isPresent()){
+            User user = dbUser.get();
+            eventCrewDTO.setUserName(user.getFirstName() + " " + (user.getLastName() != null? user.getLastName(): "") );
```
Comment (CarlosSia) [inline src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java]: if user dont have last name will become first name and empty space behind

File: src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java (hunk 1 above)
Comment (junzai97) [reply src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java]: Resolved in dba827f85e1232353deb31e387ebb6ee1242cc45

File: src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java (hunk 2 above)
Comment (junzai97) [reply src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java]: Resolved in dba827f85e1232353deb31e387ebb6ee1242cc45

File: src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java [hunk 3]
```diff
@@ -85,6 +121,38 @@ public EventCrewDTO save(EventCrewDTO eventCrewDTO) {
...
+    /**
+     * Assign user details(name and contact number) to the eventCrewDTO.
+     *
+     * @param eventCrewDTO to add on the user details' prop.
+     */
+    private EventCrewDTO mapUserDetails(EventCrewDTO eventCrewDTO) {
+        Optional<User> dbUser = userRepository.findById(eventCrewDTO.getUserId());
+        if(dbUser.isPresent()){
+            User user = dbUser.get();
+            eventCrewDTO.setUserName(user.getFirstName() + (user.getLastName() != null ? " " + user.getLastName(): "") );
```
Comment (kachun333) [inline src/main/java/com/thirdcc/webapp/service/impl/EventCrewServiceImpl.java]: this is not readable. Please break it up
```
String lastName = (user.getLastName() == null) ? "" : " " + user.getLastName();
String userName = user.getFirstName() + lastName;
eventCrewDTO.setUserName(userName);
```

**Review by kachun333 — DISMISSED** [review]

**Review by CarlosSia — APPROVED** [review]

**Review by LUXIANZE — APPROVED** [review]

**Pull request merged**